src/App.tsx

```diff
@@ -8,7 +8,65 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { Todo } from './types/Todo';
+import { FilterBy } from './types/FilterBy';
+import { TodoStatus } from './types/TodoStatus';
+
+import { getTodos } from './api';
+
+const getFilteredTodos = (todos: Todo[], query: string, filterBy: FilterBy) => {
+  let newTodos = [...todos];
+
+  if (query) {
+    const newQuery = query.toLowerCase().trim();
+
+    newTodos = newTodos.filter(todo => todo.title.toLowerCase().includes(newQuery));
+  }
+
+  switch (filterBy) {
+    case TodoStatus.Active:
+      return newTodos.filter(todo => !todo.completed);
+
+    case TodoStatus.Completed:
+      return newTodos.filter(todo => todo.completed);
+
+    default:
+      break;
+  }
+
+  return newTodos;
+};
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [filterBy, setFilterBy] = useState<FilterBy>('all');
```

Comment:
You have enum with all filter types
```suggestion
  const [filterBy, setFilterBy] = useState<FilterBy>(TodoStatus.All);
```